packages/amazon-bedrock/src/bedrock-api-types.ts

```diff
@@ -0,0 +1,122 @@
+import { Resolvable } from '@ai-sdk/provider-utils';
+
+export interface BedrockConverseInput {
+  system?: Array<{ text: string }>;
+  messages: Array<{
+    role: string;
+    content: Array<BedrockContentBlock>;
+  }>;
+  toolConfig?: BedrockToolConfiguration;
+  inferenceConfig?: {
+    maxTokens?: number;
+    temperature?: number;
+    topP?: number;
+    stopSequences?: string[];
+  };
+  additionalModelRequestFields?: Record<string, any>;
+  guardrailConfig?:
+    | BedrockGuardrailConfiguration
+    | BedrockGuardrailStreamConfiguration
+    | undefined;
+}
+
+export interface BedrockGuardrailConfiguration {
+  guardrails?: Array<{
+    name: string;
+    description?: string;
+    parameters?: Record<string, any>;
+  }>;
+}
+
+export type BedrockGuardrailStreamConfiguration = BedrockGuardrailConfiguration;
+
+export interface BedrockToolInputSchema {
+  json: Record<string, any>;
+}
+
+export interface BedrockTool {
+  toolSpec: {
+    name: string;
+    description?: string;
+    inputSchema: { json: any };
```

Comment:
Is `any` required or can `unknown` be used here?

Reply:
`JSONObject` looks viable.